### PR TITLE
constant gr legendbox size

### PR DESCRIPTION
**Pull request by isentropic**
This is a followup to https://github.com/JuliaPlots/Plots.jl/issues/3851
Setting constant scaling factor renders:
`plot(rand(10), size=(100,400), )`
image
vs old
image

### CODE REVIEW

**Comment by isentropic** [comment]
With such change, for example https://docs.juliaplots.org/latest/layouts/#Advanced-Layouts
looks much better:
```
l = @layout [
    a{0.3w} [grid(3,3)
             b{0.2h}  ]
]
plot(
    rand(10, 11),
    layout = l, legend = true, seriestype = [:bar :scatter :path],
    title = ["($i)" for j in 1:1, i in 1:11], titleloc = :right, titlefont = font(8)
)
```
image

vs old
image

**Comment by isentropic** [comment]
By reducing legendfontsize, it looks really good now I would argue

**Comment by BeastyBlacksmith** [comment]
Needs new reference images though

**Comment by t-bltg** [comment]
I agree that in the first examples (https://github.com/JuliaPlots/Plots.jl/pull/3855#issue-1010663664) it looks better.

However, in the second (https://github.com/JuliaPlots/Plots.jl/pull/3855#issuecomment-929970069), the data is unreadable.
It is IMO more important to be able to interpret the data than to see the label.

I believe the old plot:
old
looks better than the new one (unusable as is):
new

**Comment by BeastyBlacksmith** [comment]
I'd say putting labels inside such a tiny plot should not be done in the first place.
We should encourage users to put the legends outside of the plot in this case (if needed).

File: src/backends/gr.jl [hunk 1]
```diff
@@ -1260,7 +1260,10 @@ function gr_get_legend_geometry(viewport_plotarea, sp)
         GR.restorestate()
     end
 
-    legend_width_factor = (viewport_plotarea[2] - viewport_plotarea[1]) / 45 # Determines the width of legend box
+    # legend_width_factor = (viewport_plotarea[2] - viewport_plotarea[1]) / 45 # Determines the width of legend box
+    # Let's prefer constant width instead
+    legend_width_factor = 0.02
```
Comment (t-bltg) [inline src/backends/gr.jl:1265]: I'm always reticent to go back to hard-coded factors, when a dynamic solution exists.

**Comment by isentropic** [comment]
@BeastyBlacksmith @t-bltg it looks very much fine if you reduce the legendfontsize. Currently however, it looks bad either way

File: src/backends/gr.jl (hunk 1 above)
Comment (isentropic) [reply src/backends/gr.jl:1265]: Here 0.02 is chosen to match the same size under default plot size.
I do not see how 45 is not hardcoded

**Comment by isentropic** [comment]
I just showed you the really bad exaggerated case of course, but on average-sized 1:1 aspect ratio plots, this is a definite improvement

**Comment by isentropic** [comment]
There must be something off with my gr installation, as can't get to pass the tests

**Comment by isentropic** [comment]
Yes I understand that, every time I do it, too many new reference images are spawned (without legend box). I expect only the reference images with the legends to change, but a lot of them change with 1e-4 tolerance, higher than default 1e-5

**Comment by isentropic** [comment]
failed to rebase correctly...

**Pull request closed without merging**